### PR TITLE
[Enhance] refine metric of TopNSorter

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Refine the `SortFilterTime` and `SortFilterRows`.

### CODE REVIEW

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2 branch-2.3

File: be/src/exec/vectorized/chunks_sorter_topn.cpp [hunk 1]
```diff
@@ -245,21 +246,28 @@ Status ChunksSorterTopn::_filter_and_sort_data(RuntimeState* state, std::pair<Pe
         // so we can only use the index of `0` as the left boundary to filter the coming input chunks into two parts, `SMALLER_THAN_MIN_OF_SEGMENT` and `INCLUDE_IN_SEGMENT`
 
         if (_merged_segment.chunk->num_rows() >= rows_to_sort) {
+            ScopedTimer<MonotonicStopWatch> timer(_sort_filter_timer);
```
Comment (liuyehcf) [inline be/src/exec/vectorized/chunks_sorter_topn.cpp:249]: use macro `SCOPED_TIMER`?

File: be/src/exec/vectorized/chunks_sorter_topn.cpp [hunk 2]
```diff
@@ -245,21 +246,28 @@ Status ChunksSorterTopn::_filter_and_sort_data(RuntimeState* state, std::pair<Pe
...
             RETURN_IF_ERROR(_merged_segment.get_filter_array(segments, rows_to_sort, filter_array, _sort_order_flag,
                                                              _null_first_flag, smaller_num, include_num));
         } else {
+            ScopedTimer<MonotonicStopWatch> timer(_sort_filter_timer);
```
Comment (liuyehcf) [inline be/src/exec/vectorized/chunks_sorter_topn.cpp:253]: use macro SCOPED_TIMER?

File: be/src/exec/vectorized/chunks_sorter_topn.cpp [hunk 3]
```diff
@@ -271,17 +279,22 @@ Status ChunksSorterTopn::_filter_and_sort_data(RuntimeState* state, std::pair<Pe
...
 
-    return _partial_sort_col_wise(state, permutations, segments, chunk_size);
+    return _partial_sort_col_wise(state, permutations, segments);
 }
 
 Status ChunksSorterTopn::_partial_sort_col_wise(RuntimeState* state, std::pair<Permutation, Permutation>& permutations,
-                                                DataSegments& segments, const size_t chunk_size) {
+                                                DataSegments& segments) {
     const size_t rows_to_sort = _get_number_of_rows_to_sort();
+    ScopedTimer<MonotonicStopWatch> timer(_sort_timer);
```
Comment (liuyehcf) [inline be/src/exec/vectorized/chunks_sorter_topn.cpp:297]: use macro SCOPED_TIMER?

File: be/src/exec/vectorized/chunks_sorter_topn.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exec/vectorized/chunks_sorter_topn.cpp:249]: fixed

File: be/src/exec/vectorized/chunks_sorter_topn.cpp (hunk 3 above)
Comment (mofeiatwork) [reply be/src/exec/vectorized/chunks_sorter_topn.cpp]: fixed

File: be/src/exec/vectorized/chunks_sorter_topn.cpp (hunk 2 above)
Comment (mofeiatwork) [reply be/src/exec/vectorized/chunks_sorter_topn.cpp]: fixed

**Review by kangkaisen — APPROVED** [review]

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by stdpain — APPROVED** [review]

**Pull request merged**